emu/src/cpu/instruction.rs

```diff
@@ -67,11 +187,77 @@ impl From<u32> for ArmModeInstruction {
         } else if op_code.get_bits(25..=27) == 0b100 {
             BlockDataTransfer
         } else if op_code.get_bits(25..=27) == 0b101 {
-            Branch
+            let is_link = op_code.get_bit(24);
+            let offset = op_code.get_bits(0..=23) << 2;
+            Branch(condition, is_link, offset)
         } else if op_code.get_bits(26..=27) == 0b01 {
-            SingleDataTransfer
+            let op_kind: OperandKind = op_code.get_bit(25).into();
```

Comment:
Thoughts for the future: maybe we should find a better name for `OperandKind` if we want to use it everywhere, because for example here it is not really an operand but the offset, but ok I guess that for now we can leave it

Reply:
Sure, we can pick something like `SecondOperandKind` or `RegisterOrImmediateOperand`

Reply:
I like the second one more, what do you think?